ir/instr.h

```diff
@@ -475,16 +475,15 @@ class MemInstr : public Instr {
 class Alloc final : public MemInstr {
   Value *size, *mul;
   unsigned align;
-  bool initially_dead;
+  bool initially_dead = false;
 public:
-  Alloc(Type &type, std::string &&name, Value &size, Value *mul, unsigned align,
-        bool initially_dead)
-    : MemInstr(type, std::move(name)), size(&size), mul(mul), align(align),
-      initially_dead(initially_dead) {}
+  Alloc(Type &type, std::string &&name, Value &size, Value *mul, unsigned align)
+    : MemInstr(type, std::move(name)), size(&size), mul(mul), align(align) {}
 
   Value& getSize() const { return *size; }
   Value* getMul() const { return mul; }
   bool initDead() const { return initially_dead; }
+  void markAsInitiallyDead() { initially_dead = true; }
```

Comment:
Previously llvm2alive.cpp's lifetime.star/end encoding was splitted into two functions: hasLifetimeStart (which checks AllocaInst's uses) and visitIntrinsicInst (which adds StartLifetime instance)
I removed hasLifetimeStart, and made visitIntrinsicInst to update this initially-dead flag if lifetime.start uses the alloca.
This `markAsInitiallyDead` function is added to implement it.